pkg/metadata/inventories/inventories.go

```diff
@@ -54,22 +54,25 @@ var (
 	timeSince = time.Since
 )
 
+// AgentMetadataName is an enum type containing all defined keys for
+// SetAgentMetadata.
+type AgentMetadataName string
+
+// Constants for the metadata names; these are defined in
+// pkg/metadata/inventories/README.md and any additions should
+// be updated there as well.
 const (
-	// CloudProviderMetatadaName is the field name to use to set the cloud
-	// provider name in the agent metadata.
-	CloudProviderMetatadaName = "cloud_provider"
-	// HostnameSourceMetadataName is the field name to use to set the hostname
-	// source in the agent metadata.
-	HostnameSourceMetadataName = "hostname_source"
+	AgentCloudProvider  AgentMetadataName = "cloud_provider"
+	AgentHostnameSource                   = "hostname_source"
```

Comment:
I don't think the type is carried over to the second decl given that you specify the value of similar declaration (it should end up as a string): https://play.golang.org/p/WSLPr95T-mW

Reply:
Indeed, I was copying integer enums and didn't catch this.  Thanks!